client/my-sites/customer-home/cards/tasks/connect-accounts.jsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { useTranslate } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { Button } from '@automattic/components';
+import { isMobile } from '@automattic/viewport';
+
+/**
+ * Internal dependencies
+ */
+import ActionPanel from 'components/action-panel';
+import ActionPanelTitle from 'components/action-panel/title';
+import ActionPanelBody from 'components/action-panel/body';
+import ActionPanelFigure from 'components/action-panel/figure';
+import ActionPanelCta from 'components/action-panel/cta';
+import Gridicon from 'components/gridicon';
+import QueryPublicizeConnections from 'components/data/query-publicize-connections';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { savePreference } from 'state/preferences/actions';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const ConnectAccountsTask = ( { skipTask, siteSlug } ) => {
+	const translate = useTranslate();
+
+	return (
+		<ActionPanel className="tasks connect-accounts">
+			<QueryPublicizeConnections selectedSite />
+			<ActionPanelBody>
+				{ ! isMobile() && (
+					<ActionPanelFigure align="right">
+						<img src="/calypso/images/stats/tasks/social-links.svg" alt="" />
+					</ActionPanelFigure>
+				) }
+				<div className="tasks__timing">
+					<Gridicon icon="time" size={ 18 } />
+					<p>{ translate( '3 minutes' ) }</p>
+				</div>
+				<ActionPanelTitle>{ translate( 'Drive traffic to your site' ) }</ActionPanelTitle>
+				<p className="tasks__description">
+					{ translate(
+						'Integrate your site with social media to automatically post your content and drive traffic to your site.'
+					) }
+				</p>
+				<ActionPanelCta>
+					<Button
+						className="tasks__action"
+						primary
+						href={
+							isMobile()
+								? `/marketing/connections/${ siteSlug }`
+								: `/marketing/connections/${ siteSlug }?tour=marketingConnectionsTour`
+						}
+					>
+						{ translate( 'Connect accounts' ) }
+					</Button>
+					<Button className="tasks__action-skip is-link" onClick={ skipTask }>
```

Comment:
🤔 Since we don't re-trigger the request to the API layout after dismissing a card, this one won't be hidden until we reload the page. While we come up with a mechanism to do this out of the box, I think we need to explicitly hide the task when triggering an action that dismiss it.

Reply:
Would you recommend using `setPreference` to temporarily store the preference and do so?

Reply:
That's already done [when using `savePreference`](https://github.com/Automattic/wp-calypso/blob/e7ff36f24b352c44261e3b339895251c9f044ea5/client/state/preferences/actions.js#L82).

I'd recommend to have a temporary visibility status in the local state. Something like this:

```
const [ isVisible, setIsVisible ] = useState( true );

if ( ! isVisible ) {
  return null;
}

return (
  <Button onClick={ () => { 
    setIsVisible( false ); 
    skipTask();
  } />
);
```

Reply:
Thank you, updated to that. :)

Reply:
It's also perfectly fine to iterate on the dismiss functionality in a follow up PR. I'm not sure yet for what happens when we only have one task or a few tasks.

Depending on what we pick for behavior, making an explicit get/post to our layout API while clicking on the secondary link is highly likely.